Danki2/Assets/Scripts/Actor/Actor.cs

```diff
@@ -22,6 +25,12 @@ void Update()
     {
         _statsManager.Rebase();
         _effectTracker.ProcessEffects();
+        this.Act();
+    }
+
+    protected virtual void Act()
```

Comment:
I think this method should be abstract and `Actor` shouldn't have an AI, it feels like `Actor` shouldn't really know about `AI`. Then inheriting classes have the choice as to whether they want to use AI to act. Then the player won't have to have to override 2 methods and it just needs to implement `Act`. I could imagine there's other stuff that we won't want to use AI (like a shopkeeper or something) and it'd be nice if they didn't have to override these methods as well. Then `Enemy` and `Friendly` can have `public abstract IAI AI { get; }` like you've got above and can use the AI to implement `Act()` like you've got here (those methods won't have to be virtual then I don't think, which is nice).